internal/pkg/gateway/endorsement.go

```diff
@@ -85,8 +92,8 @@ func (p *plan) endorsers() []*endorser {
 
 // Invoke update when an endorsement has been successfully received for the given endorser.
 // All layouts containing the group that contains this endorser are updated with the endorsement.
-// Returns array of endorsements, if at least one layout in the plan has been satisfied, otherwise nil.
-func (p *plan) update(endorser *endorser, endorsement *peer.ProposalResponse) []*peer.ProposalResponse {
+// Returns a ChaincodeEndorsedAction, if at least one layout in the plan has been satisfied, otherwise nil.
+func (p *plan) update(endorser *endorser, response *peer.ProposalResponse) (*peer.ChaincodeEndorsedAction, error) {
```

Comment:
update() function is not introduced in this PR, but I've always thought the function name could be more descriptive, something like processEndorsement().